docker/dockerfile-python

```diff
@@ -1,21 +1,28 @@
 FROM ubuntu:16.04
 
 RUN apt-get update && \
-    apt-get install -y cmake build-essential gcc g++ git wget && \
+    apt-get install -y --no-install-recommends \
+        cmake \
+        build-essential \
+        gcc \
+        g++ \
+        git \
+        wget && \
 
 # python-package
     # miniconda
     wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
     /bin/bash Miniconda3-latest-Linux-x86_64.sh -f -b -p /opt/conda && \
     export PATH="/opt/conda/bin:$PATH" && \
+    conda config --set always_yes yes --set changeps1 no && \
     # lightgbm
-    conda install -y numpy scipy scikit-learn pandas && \
-    git clone --recursive https://github.com/Microsoft/LightGBM && \
+    conda install -q -y numpy scipy scikit-learn pandas && \
+    git clone --recursive --branch stable https://github.com/Microsoft/LightGBM && \
```

Comment:
could you add `--depth 1`? For the use cases these Dockerfiles are intended for, it's probably unnecessary to have the entire history.

This would slightly reduce the size of the resulting image.